### PR TITLE
Implement build steps using FPM-Cookery

### PR DESCRIPTION
**Pull request by shelbyspees**

### CODE REVIEW

**Comment by shelbyspees** [comment]
I made the PR at this early stage as a sanity check to make sure I'm on the right track identifying what happens where. Updates coming soon.

File: lib/revolution/travis_ci.rb [hunk 1]
```diff
@@ -11,10 +11,37 @@ def self.current_branch
...
+  end
+
+  def self.run_type
+    pull_request? ? :pr : :merge
+  end
+
+  def self.changes
+    # TODO
+  end
 end
```
Comment (shelbyspees) [inline lib/revolution/travis_ci.rb]: Moving travis_ci module changes to a different PR

**Comment by shelbyspees** [comment]
I'm not exactly sure how to use fpm-cookery's Packager class. I'm getting an error that says the `recipe.config` method has to be defined before the recipe can be loaded. See this gist: https://gist.github.com/shelbyspees/6eeef869c9258cac11f1da63dfd7b910

File: lib/revolution/build.rb [hunk 2]
```diff
@@ -2,9 +2,34 @@
 # frozen_string_literal: true
 
 module Build
-  def self.build_package(name = nil, location = nil)
-    puts '[Inside Build.build_package]'
-    puts "Building package: #{name} from #{location}..."
-    puts "Package #{name} successfully built!"
+  def self.log(pkg_name)
+    log_dir = "recipes/#{pkg_name}/log"
```
Comment (smaslennikov) [inline lib/revolution/build.rb]: In the future, this needs to be fixed: Travis will expect package dirs to be in `./` (since we'll be running the tool from rpm-packages itself), while local dev env will expect them in `/vagrant/recipes`. 

A #TODO noting this will suffice for now.

File: lib/revolution/build.rb [hunk 3]
```diff
@@ -2,9 +2,34 @@
...
+  def self.run(pkg_name, command)
+    fpm_cook = 'fpm-cook package --no-deps' if command == 'build'
+    fpm_cook = 'fpm-cook clean' if command == 'clean'
+
+    log_dir = log(pkg_name)
+    outfile = "#{log_dir}/#{Time.now.strftime('%F_%T')}"
+    output  = File.open(outfile, 'w')
+
+    pid = Process.spawn("#{fpm_cook} recipes/#{pkg_name}/recipe.rb",
+                        out: '/dev/null', err: output)
```
Comment (smaslennikov) [inline lib/revolution/build.rb]: The logic of rerouting out and err should probably be specified via verbose/debug arguments. If you're not ready to implement that now, please add them as #TODOs.

File: spec/build_spec.rb [hunk 4]
```diff
@@ -2,13 +2,56 @@
...
 require 'rspec'
+require 'yaml'
 require 'revolution/build'
 
 describe 'Build' do
-  describe '.build_package' do
-    context 'with no arguments' do
-      it 'returns nil' do
-        expect(Build.build_package).to be nil
+  context 'passing in grindr-base with recipes/ directory mounted' do
```
Comment (smaslennikov) [inline spec/build_spec.rb]: See above, we shouldn't depend on recipes being _within_ revolution.

File: lib/revolution/build.rb (hunk 3 above)
Comment (shelbyspees) [reply lib/revolution/build.rb]: I made a ticket for this but I'll add a TODO

File: lib/revolution/build.rb (hunk 2 above)
Comment (shelbyspees) [reply lib/revolution/build.rb]: Added TODO

File: spec/build_spec.rb (hunk 4 above)
Comment (shelbyspees) [reply spec/build_spec.rb]: Added TODO
